front/src/modules/timesStops/hooks/useOutputTableData.ts

```diff
@@ -65,51 +73,61 @@ function useOutputTableData(
     (opPoint) => `${opPoint.name}-${opPoint.ch}`
   );
 
-  const outputTableData = useMemo(
-    () =>
-      suggestedOperationalPoints.map((sugOpPoint, sugOpIndex) => {
-        const opPoint = operationPointsByNameCh[`${sugOpPoint.name}-${sugOpPoint.ch}`];
-        if (!opPoint) {
-          return sugOpPoint;
-        }
-        const nextOpPoint = findNextScheduledOpPoint(
-          operationalPoints,
-          pathStepsWithOpPointIndices,
-          sugOpIndex
-        );
-        const pathStepKey = `${sugOpPoint.uic}-${sugOpPoint.ch}`;
+  const outputTableData = useMemo(() => {
+    const pathStepRows = pathStepsWithPositionOnPath.map((pathStep) => {
+      const schedule = scheduleByAt[pathStep.id];
+      const scheduleData = computeScheduleData(schedule, selectedTrainSchedule.start_time);
+      return { ...pathStep, ...formatScheduleData(scheduleData) };
+    });
 
-        if (pathStepKey in pathStepsByUic) {
-          const pathStepId = pathStepsByUic[pathStepKey].id || '';
-          const schedule = scheduleByAt[pathStepId];
-          const scheduleData = computeScheduleData(schedule, selectedTrainSchedule.start_time);
-          const formattedScheduleData = formatScheduleData(scheduleData);
-          const marginsData = nextOpPoint
-            ? computeMargins(
-                simulatedTrain,
-                opPoint,
-                nextOpPoint,
-                selectedTrainSchedule,
-                pathStepId
-              )
-            : null;
-          const calculatedArrival = checkAndFormatCalculatedArrival(scheduleData, opPoint.time);
+    const suggestedOpRows = suggestedOperationalPoints.map((sugOpPoint, sugOpIndex) => {
+      const opPoint = operationPointsByNameCh[`${sugOpPoint.name}-${sugOpPoint.ch}`];
+      if (!opPoint) {
+        return sugOpPoint;
+      }
+      const nextOpPoint = findNextScheduledOpPoint(
+        operationalPoints,
+        pathStepsWithOpPointIndices,
+        sugOpIndex
+      );
+      const pathStepKey = `${sugOpPoint.uic}-${sugOpPoint.ch}`;
 
-          return {
-            ...sugOpPoint,
-            ...formattedScheduleData,
-            onStopSignal: schedule?.on_stop_signal || '',
-            calculatedArrival,
-            calculatedDeparture:
-              opPoint.duration > 0 ? secToHoursString(opPoint.time + opPoint.duration, true) : '',
-            ...marginsData,
-          } as SuggestedOP;
-        }
+      if (pathStepKey in pathStepsByUic) {
+        const pathStepId = pathStepsByUic[pathStepKey].id || '';
```

Comment:
Any reason why we're still using the uic+ch here to fish the `pathStepId` (and then the `schedule`) instead of using a logic to the one below?

Reply:
I didn't change the logic here. I can leave it like this for now if it's ok for you ?

Reply:
Sure